hudi-common/src/main/java/org/apache/hudi/common/table/view/AbstractTableFileSystemView.java

```diff
@@ -915,7 +958,16 @@ public final Option<FileSlice> getLatestFileSlice(String partitionStr, String fi
         if (!fs.isPresent()) {
           return Option.empty();
         }
-        return Option.ofNullable(filterUncommittedFiles(fs.get(), true).map(this::addBootstrapBaseFileIfPresent).findFirst().orElse(null));
+        HoodieTableVersion tableVersion = metaClient.getTableConfig().getTableVersion();
+        Stream<FileSlice> fileSlices = tableVersion.greaterThanOrEquals(HoodieTableVersion.EIGHT)
```

Comment:
Maybe we just maintain a flag variable `isTableVersionAboveEight` so that the code would be cleaner.